Build fails with System.IO.FileNotFoundException

- **What version of the product are you using?**
    - Master at 6602977496ea4aed649b8e5a3273331a01a46865

- **What architecture x86 or x64?**
    - Applies to both

- **On what operating system?**
    - Windows 7

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
    - Applies to all

- **What steps will reproduce the problem?**
    - Clone a clean version of the repo
    - Restore all nuget packages
    - Build Solution

- **What is the expected output? What do you see instead?**
    - Build should succeed instead it fails with this message:
```
9>------ Build started: Project: CefSharp.Wpf.Example, Configuration: Debug x86 ------
10>------ Build started: Project: CefSharp.WinForms.Example, Configuration: Debug x86 ------
11>------ Build started: Project: CefSharp.OffScreen.Example, Configuration: Debug x86 ------
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error MSB3883: Unexpected exception: 
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : System.IO.FileNotFoundException: Could not load file or assembly 'System.Security.Principal.Windows, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : File name: 'System.Security.Principal.Windows, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Security.Principal.Windows, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : File name: 'System.Security.Principal.Windows, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.AssemblyResolution.LoadAssemblyWithRedirects(AssemblyName name)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.AssemblyResolution.ResolveAssembly(String assemblyDisplayName, Assembly requestingAssemblyOpt)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Roslyn.Utilities.CorLightup.Desktop.AssemblyResolveWrapper.Stub(Object sender, Object resolveEventArgs)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : WRN: Assembly binding logging is turned OFF.
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : Note: There is some performance penalty associated with assembly bind failure logging.
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.CommandLine.BuildServerConnection.GetPipeNameForPathOpt(String compilerExeDirectory)
11>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.ManagedCompiler.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error MSB3883: Unexpected exception: 
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : System.IO.FileNotFoundException: Could not load file or assembly 'System.Security.Principal.Windows, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : File name: 'System.Security.Principal.Windows, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Security.Principal.Windows, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : File name: 'System.Security.Principal.Windows, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.AssemblyResolution.LoadAssemblyWithRedirects(AssemblyName name)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.AssemblyResolution.ResolveAssembly(String assemblyDisplayName, Assembly requestingAssemblyOpt)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Roslyn.Utilities.CorLightup.Desktop.AssemblyResolveWrapper.Stub(Object sender, Object resolveEventArgs)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : WRN: Assembly binding logging is turned OFF.
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : Note: There is some performance penalty associated with assembly bind failure logging.
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.CommandLine.BuildServerConnection.GetPipeNameForPathOpt(String compilerExeDirectory)
10>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.ManagedCompiler.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands)
9>CefSharp\CefSharp.Wpf.Example\App.xaml.cs(34,17,34,38): warning CS0162: Unreachable code detected
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error MSB3883: Unexpected exception: 
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : System.IO.FileNotFoundException: Could not load file or assembly 'System.Security.Principal.Windows, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : File name: 'System.Security.Principal.Windows, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Security.Principal.Windows, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : File name: 'System.Security.Principal.Windows, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.AssemblyResolution.LoadAssemblyWithRedirects(AssemblyName name)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.AssemblyResolution.ResolveAssembly(String assemblyDisplayName, Assembly requestingAssemblyOpt)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Roslyn.Utilities.CorLightup.Desktop.AssemblyResolveWrapper.Stub(Object sender, Object resolveEventArgs)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : WRN: Assembly binding logging is turned OFF.
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : Note: There is some performance penalty associated with assembly bind failure logging.
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.CommandLine.BuildServerConnection.GetPipeNameForPathOpt(String compilerExeDirectory)
9>CefSharp\packages\Microsoft.Net.Compilers.2.7.0\tools\Microsoft.CSharp.Core.targets(84,5): error : at Microsoft.CodeAnalysis.BuildTasks.ManagedCompiler.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands)
```

- **Please provide any additional information below.**
```
Microsoft Visual Studio Community 2017
Version 15.8.3
VisualStudio.15.Release/15.8.3+28010.2019
Microsoft .NET Framework
Version 4.7.02558
```

--- a/CefSharp.OffScreen.Example/CefSharp.OffScreen.Example.csproj
+++ b/CefSharp.OffScreen.Example/CefSharp.OffScreen.Example.csproj
@@ -67,6 +67,9 @@
     <Reference Include="System.Core" />
     <Reference Include="System.Drawing" />
     <Reference Include="Microsoft.CSharp" />
+    <Reference Include="System.Security.Principal.Windows, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Security.Principal.Windows.4.5.0\lib\net461\System.Security.Principal.Windows.dll</HintPath>
+    </Reference>
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Program.cs" />

--- a/CefSharp.OffScreen.Example/CefSharp.OffScreen.Example.csproj
+++ b/CefSharp.OffScreen.Example/CefSharp.OffScreen.Example.csproj
@@ -67,9 +67,6 @@
     <Reference Include="System.Core" />
     <Reference Include="System.Drawing" />
     <Reference Include="Microsoft.CSharp" />
-    <Reference Include="System.Security.Principal.Windows, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\packages\System.Security.Principal.Windows.4.5.0\lib\net461\System.Security.Principal.Windows.dll</HintPath>
-    </Reference>
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Program.cs" />

--- a/CefSharp.OffScreen.Example/CefSharp.OffScreen.Example.csproj
+++ b/CefSharp.OffScreen.Example/CefSharp.OffScreen.Example.csproj
@@ -1,8 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props" Condition="Exists('..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props')" />
   <Import Project="..\packages\cef.redist.x86.3.3497.1827\build\cef.redist.x86.props" Condition="Exists('..\packages\cef.redist.x86.3.3497.1827\build\cef.redist.x86.props')" />
   <Import Project="..\packages\cef.redist.x64.3.3497.1827\build\cef.redist.x64.props" Condition="Exists('..\packages\cef.redist.x64.3.3497.1827\build\cef.redist.x64.props')" />
-  <Import Project="..\packages\Microsoft.Net.Compilers.2.7.0\build\Microsoft.Net.Compilers.props" Condition="Exists('..\packages\Microsoft.Net.Compilers.2.7.0\build\Microsoft.Net.Compilers.props')" />
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
@@ -101,4 +101,10 @@
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <Import Project="$(SolutionDir)\CefSharp.props" />
   <Import Project="$(SolutionDir)\CefSharp.Native.props" />
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+    </PropertyGroup>
+    <Error Condition="!Exists('..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props'))" />
+  </Target>
 </Project>

--- a/CefSharp.OffScreen.Example/packages.config
+++ b/CefSharp.OffScreen.Example/packages.config
@@ -3,4 +3,5 @@
   <package id="cef.redist.x64" version="3.3497.1827" targetFramework="net462" />
   <package id="cef.redist.x86" version="3.3497.1827" targetFramework="net462" />
   <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
+  <package id="System.Security.Principal.Windows" version="4.5.0" targetFramework="net462" />
 </packages>

--- a/CefSharp.OffScreen.Example/packages.config
+++ b/CefSharp.OffScreen.Example/packages.config
@@ -3,5 +3,4 @@
   <package id="cef.redist.x64" version="3.3497.1827" targetFramework="net462" />
   <package id="cef.redist.x86" version="3.3497.1827" targetFramework="net462" />
   <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
-  <package id="System.Security.Principal.Windows" version="4.5.0" targetFramework="net462" />
 </packages>

--- a/CefSharp.OffScreen.Example/packages.config
+++ b/CefSharp.OffScreen.Example/packages.config
@@ -2,5 +2,5 @@
 <packages>
   <package id="cef.redist.x64" version="3.3497.1827" targetFramework="net462" />
   <package id="cef.redist.x86" version="3.3497.1827" targetFramework="net462" />
-  <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
+  <package id="Microsoft.Net.Compilers" version="2.9.0" targetFramework="net462" developmentDependency="true" />
 </packages>

--- a/CefSharp.WinForms.Example/CefSharp.WinForms.Example.csproj
+++ b/CefSharp.WinForms.Example/CefSharp.WinForms.Example.csproj
@@ -1,8 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props" Condition="Exists('..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props')" />
   <Import Project="..\packages\cef.redist.x86.3.3497.1827\build\cef.redist.x86.props" Condition="Exists('..\packages\cef.redist.x86.3.3497.1827\build\cef.redist.x86.props')" />
   <Import Project="..\packages\cef.redist.x64.3.3497.1827\build\cef.redist.x64.props" Condition="Exists('..\packages\cef.redist.x64.3.3497.1827\build\cef.redist.x64.props')" />
-  <Import Project="..\packages\Microsoft.Net.Compilers.2.7.0\build\Microsoft.Net.Compilers.props" Condition="Exists('..\packages\Microsoft.Net.Compilers.2.7.0\build\Microsoft.Net.Compilers.props')" />
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
@@ -189,4 +189,10 @@
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <Import Project="$(SolutionDir)\CefSharp.props" />
   <Import Project="$(SolutionDir)\CefSharp.Native.props" />
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+    </PropertyGroup>
+    <Error Condition="!Exists('..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props'))" />
+  </Target>
 </Project>

--- a/CefSharp.WinForms.Example/CefSharp.WinForms.Example.csproj
+++ b/CefSharp.WinForms.Example/CefSharp.WinForms.Example.csproj
@@ -62,6 +62,9 @@
     <Reference Include="System.Data" />
     <Reference Include="System.Drawing" />
     <Reference Include="System.Security" />
+    <Reference Include="System.Security.Principal.Windows, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Security.Principal.Windows.4.5.0\lib\net461\System.Security.Principal.Windows.dll</HintPath>
+    </Reference>
     <Reference Include="System.Windows.Forms" />
     <Reference Include="System.Xml" />
   </ItemGroup>

--- a/CefSharp.WinForms.Example/CefSharp.WinForms.Example.csproj
+++ b/CefSharp.WinForms.Example/CefSharp.WinForms.Example.csproj
@@ -62,9 +62,6 @@
     <Reference Include="System.Data" />
     <Reference Include="System.Drawing" />
     <Reference Include="System.Security" />
-    <Reference Include="System.Security.Principal.Windows, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\packages\System.Security.Principal.Windows.4.5.0\lib\net461\System.Security.Principal.Windows.dll</HintPath>
-    </Reference>
     <Reference Include="System.Windows.Forms" />
     <Reference Include="System.Xml" />
   </ItemGroup>

--- a/CefSharp.WinForms.Example/packages.config
+++ b/CefSharp.WinForms.Example/packages.config
@@ -3,4 +3,5 @@
   <package id="cef.redist.x64" version="3.3497.1827" targetFramework="net462" />
   <package id="cef.redist.x86" version="3.3497.1827" targetFramework="net462" />
   <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
+  <package id="System.Security.Principal.Windows" version="4.5.0" targetFramework="net462" />
 </packages>

--- a/CefSharp.WinForms.Example/packages.config
+++ b/CefSharp.WinForms.Example/packages.config
@@ -3,5 +3,4 @@
   <package id="cef.redist.x64" version="3.3497.1827" targetFramework="net462" />
   <package id="cef.redist.x86" version="3.3497.1827" targetFramework="net462" />
   <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
-  <package id="System.Security.Principal.Windows" version="4.5.0" targetFramework="net462" />
 </packages>

--- a/CefSharp.WinForms.Example/packages.config
+++ b/CefSharp.WinForms.Example/packages.config
@@ -2,5 +2,5 @@
 <packages>
   <package id="cef.redist.x64" version="3.3497.1827" targetFramework="net462" />
   <package id="cef.redist.x86" version="3.3497.1827" targetFramework="net462" />
-  <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
+  <package id="Microsoft.Net.Compilers" version="2.9.0" targetFramework="net462" developmentDependency="true" />
 </packages>

--- a/CefSharp.Wpf.Example/CefSharp.Wpf.Example.csproj
+++ b/CefSharp.Wpf.Example/CefSharp.Wpf.Example.csproj
@@ -91,6 +91,9 @@
       <RequiredTargetFramework>3.5</RequiredTargetFramework>
     </Reference>
     <Reference Include="System.Drawing" />
+    <Reference Include="System.Security.Principal.Windows, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Security.Principal.Windows.4.5.0\lib\net461\System.Security.Principal.Windows.dll</HintPath>
+    </Reference>
     <Reference Include="System.Windows.Interactivity, Version=4.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\packages\MvvmLightLibs.5.1.1.0\lib\net45\System.Windows.Interactivity.dll</HintPath>
     </Reference>

--- a/CefSharp.Wpf.Example/CefSharp.Wpf.Example.csproj
+++ b/CefSharp.Wpf.Example/CefSharp.Wpf.Example.csproj
@@ -91,9 +91,6 @@
       <RequiredTargetFramework>3.5</RequiredTargetFramework>
     </Reference>
     <Reference Include="System.Drawing" />
-    <Reference Include="System.Security.Principal.Windows, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\packages\System.Security.Principal.Windows.4.5.0\lib\net461\System.Security.Principal.Windows.dll</HintPath>
-    </Reference>
     <Reference Include="System.Windows.Interactivity, Version=4.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\packages\MvvmLightLibs.5.1.1.0\lib\net45\System.Windows.Interactivity.dll</HintPath>
     </Reference>

--- a/CefSharp.Wpf.Example/CefSharp.Wpf.Example.csproj
+++ b/CefSharp.Wpf.Example/CefSharp.Wpf.Example.csproj
@@ -1,8 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props" Condition="Exists('..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props')" />
   <Import Project="..\packages\cef.redist.x86.3.3497.1827\build\cef.redist.x86.props" Condition="Exists('..\packages\cef.redist.x86.3.3497.1827\build\cef.redist.x86.props')" />
   <Import Project="..\packages\cef.redist.x64.3.3497.1827\build\cef.redist.x64.props" Condition="Exists('..\packages\cef.redist.x64.3.3497.1827\build\cef.redist.x64.props')" />
-  <Import Project="..\packages\Microsoft.Net.Compilers.2.7.0\build\Microsoft.Net.Compilers.props" Condition="Exists('..\packages\Microsoft.Net.Compilers.2.7.0\build\Microsoft.Net.Compilers.props')" />
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
@@ -194,4 +194,10 @@
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <Import Project="$(SolutionDir)\CefSharp.props" />
   <Import Project="$(SolutionDir)\CefSharp.Native.props" />
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+    </PropertyGroup>
+    <Error Condition="!Exists('..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Net.Compilers.2.9.0\build\Microsoft.Net.Compilers.props'))" />
+  </Target>
 </Project>

--- a/CefSharp.Wpf.Example/packages.config
+++ b/CefSharp.Wpf.Example/packages.config
@@ -5,4 +5,5 @@
   <package id="CommonServiceLocator" version="1.3" targetFramework="net45" />
   <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
   <package id="MvvmLightLibs" version="5.1.1.0" targetFramework="net45" />
+  <package id="System.Security.Principal.Windows" version="4.5.0" targetFramework="net462" />
 </packages>

--- a/CefSharp.Wpf.Example/packages.config
+++ b/CefSharp.Wpf.Example/packages.config
@@ -3,6 +3,6 @@
   <package id="cef.redist.x64" version="3.3497.1827" targetFramework="net462" />
   <package id="cef.redist.x86" version="3.3497.1827" targetFramework="net462" />
   <package id="CommonServiceLocator" version="1.3" targetFramework="net45" />
-  <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
+  <package id="Microsoft.Net.Compilers" version="2.9.0" targetFramework="net462" developmentDependency="true" />
   <package id="MvvmLightLibs" version="5.1.1.0" targetFramework="net45" />
 </packages>

--- a/CefSharp.Wpf.Example/packages.config
+++ b/CefSharp.Wpf.Example/packages.config
@@ -5,5 +5,4 @@
   <package id="CommonServiceLocator" version="1.3" targetFramework="net45" />
   <package id="Microsoft.Net.Compilers" version="2.7.0" targetFramework="net452" developmentDependency="true" />
   <package id="MvvmLightLibs" version="5.1.1.0" targetFramework="net45" />
-  <package id="System.Security.Principal.Windows" version="4.5.0" targetFramework="net462" />
 </packages>

